proto/api/v2/compliance_results_service.proto

```diff
@@ -108,6 +121,27 @@ service ComplianceResultsService {
         };
     }
 
+    // GetComplianceScanProfiles fetches all profiles related to a scan
+    // Supported RawQuery query fields
+    //   - scan: id of the compliance scan (require(scan or cluster))
+    //   - cluster: id of the cluster (require(scan or cluster))
+    rpc GetComplianceScanProfiles (RawQuery) returns (ListComplianceProfilesResponse) {
+        option (google.api.http) = {
+            get: "/v2/compliance/scan/profiles"
+        };
+    }
+
+    // GetComplianceScanProfileResults retrieves the most recent compliance operator scan results for the specified query
+    // Supported RawQuery query fields:
+    //   - profile: id of the profile (required)
+    //   - scan: id of the compliance scan (optional)
+    //   - cluster: id of the cluster (optional)
+    rpc GetComplianceScanProfileResults (RawQuery) returns (ComplianceScanResult) {
```

Comment:
What does this endpoint provide us that you cannot get from `rpc GetComplianceScanResults (RawQuery) returns (ListComplianceScanResultsResponse)`?  That endpoint should provide a list of 1 for what I think you are trying to achieve here.  Am I off base?

Reply:
No that's a good point. Might provide more clarity but definitely over redundant.

Reply:
Looking at it more would we be able to just use `GetComplianceProfileScanStats` instead of `GetComplianceScanProfiles` for getting the list of profiles on the results page? I ask because I'd need a profiles "compliance percentage" if I'm following the mocks. We'd need to be able to query based off a scan though.

Reply:
if my memory serves, the intent of `GetComplianceScanProfiles` was to provide an overview for the profiles.  Each scan that executed that profile is included.  So we could query that by scan name and/or profile.  When I made these I went through the subset powerpoint to try to make sure I covered each piece based on how I would do it.  So I'm open to suggestions.  

Regarding the percentage, `ComplianceScanStatsShim` provides total and passing for each scan within the group.  Does the backend need to do the math to get the percentage?  Are you looking for overall numbers in `ComplianceProfileScanStats` for all the scans in that profile?

Reply:
Yeah `GetComplianceScanProfiles` will definitely be used on the dashboard to get an overall snapshot of profiles compliance but if it can be queried based off scan/cluster/profile then we would be able to use it on the reports page as well. In which case `GetComplianceScanProfiles` wouldn't be needed.

![Screenshot 2023-09-18 at 11 11 51 AM](https://github.com/stackrox/stackrox/assets/61400697/a08deaf8-fe08-4c93-81b8-31b7d5d216ce)


Getting the percentage in the front end isn't a concern.